src/modules/logicModule/LM_PC_KPIRewarder_v1.sol

```diff
@@ -284,6 +284,35 @@ contract LM_PC_KPIRewarder_v1 is
         IERC20(stakingToken).safeTransferFrom(sender, address(this), amount);
     }
 
+    function deleteStuckAssertion(bytes32 assertionId)
+        public
+        onlyOrchestratorAdmin
+    {
+        // Ensure the assertionId exists in this contract (since malicious assertions could callback this contract)
+        if (assertionData[assertionId].dataId == bytes32(0x0)) {
+            revert Module__LM_PC_KPIRewarder_v1__NonExistentAssertionId(
+                assertionId
+            );
+        }
+
+        uint assertionExpirationTime =
+            oo.getAssertion(assertionId).expirationTime;
+
+        if (block.timestamp <= assertionExpirationTime) {
+            revert Module__LM_PC_KPIRewarder_v1__AssertionNotStuck(assertionId);
+        }
+
+        try oo.settleAssertion(assertionId) {
+            // If the assertion can be settled, it doesn't qualify as stuck and we revert
+            revert Module__LM_PC_KPIRewarder_v1__AssertionNotStuck(assertionId);
```

Comment:
I think it'd be better to just let it succeed here and settle the assertion instead of reverting

Reply:
This comment was included in our session and we decided to leave it as is. Reason being that the function call expects to delete assertions which are stuck. When the assumption of the user calling the function is wrong, ie. it is not stuck, then the user would be inform by this through the revert.

If we would let it success then we might do something that is unintended, and not part of the users aim in calling the function to delete stuck assertion

Reply:
Ok. Just to mention in general a malicious token could still make the proposal stuck, but I think the oracle has a whitelist for currencies so it's ok.